agrona/src/main/java/org/agrona/ExpandableArrayBuffer.java

```diff
@@ -1269,13 +1294,50 @@ public int putNaturalLongAscii(final int index, final long value)
         ensureCapacity(index, length);
 
         long quotient = value;
-        while (i >= 0)
+        final byte[] dest = byteArray;
+        while (quotient >= 100000000)
+        {
+            final int lastEightDigits = (int)(quotient % 100000000);
+            quotient /= 100000000;
+
+            final int upperPart = lastEightDigits / 10000;
+            final int lowerPart = lastEightDigits % 10000;
+
+            final int u1 = (upperPart / 100) << 1;
+            final int u2 = (upperPart % 100) << 1;
+            final int l1 = (lowerPart / 100) << 1;
+            final int l2 = (lowerPart % 100) << 1;
+
+            i -= 8;
+
+            dest[index + i + 1] = ASCII_DIGITS[u1];
+            dest[index + i + 2] = ASCII_DIGITS[u1 + 1];
+            dest[index + i + 3] = ASCII_DIGITS[u2];
+            dest[index + i + 4] = ASCII_DIGITS[u2 + 1];
+            dest[index + i + 5] = ASCII_DIGITS[l1];
+            dest[index + i + 6] = ASCII_DIGITS[l1 + 1];
+            dest[index + i + 7] = ASCII_DIGITS[l2];
+            dest[index + i + 8] = ASCII_DIGITS[l2 + 1];
+        }
+
+        while (quotient >= 100)
```

Comment:
Hi!

The code starting here seems to be the same as:
https://github.com/real-logic/agrona/pull/236/files#diff-2476f7abe13cfde689b9e83cb89c9f82285bc252ad1ce080e97ccc9b2033e92cR1219

Would it be worse if it was a method instead?